src/builders/enzymes.jl

```diff
@@ -113,3 +131,217 @@ function gene_product_isozyme_constraints(
+            maybeget(reaction_isozymes, string(rid), string(isozyme)),
+        ),
+    ) *
+    :gene_product_capacity^(
+        capacity isa Float64 ?
+        C.Constraint(
+            value = sum(
+                gpa.value * gene_product_molar_masses[String(gp)] for
+                (gp, gpa) in gene_product_amounts
+            ),
+            bound = C.Between(0, capacity),
+        ) :
+        C.ConstraintTree(
+            Symbol(id) => C.Constraint(
+                value = sum(
+                    gene_product_amounts[Symbol(gp)].value *
+                    gene_product_molar_masses[gp] for gp in gps
+                ),
+                bound = C.Between(0, limit),
+            ) for (id, gps, limit) in capacity_limits
+        )
+    )
+end
+
+export enzyme_constraints
+
+"""
+$(TYPEDSIGNATURES)
+
+Returns simplified enzyme constraints using the variables: `fluxes`,
+`fluxes_reverse`, and `fluxes_forward`. These variables should already be
+present in the model, see [`unidirectional_fluxes`](@ref) for a quick way to add
+the latter two variables to a normal flux balance model.
+
+Only reactions in `reaction_isozymes`, which is a mapping of reaction
+identifiers to [`Isozyme`](@ref) descriptions are included in the constraints.
+For each gene product in `reaction_isozymes`, a corresponding entry in
+`gene_product_molar_masses` must be present, which is a mapping of gene products
+to their molar masses. Internally, the cheapest and fastest isozyme (minimum of
+MW/kcat for each isozyme) is used in the capacity bound.
+
+`capacity` may be a single number, which sets the limit of protein required for
+all the fluxes in `reaction_isozymes` (mass/gDW units). Alternatively,
+`capacity` may be a vector of identifier-fluxes-limit triples that make a
+constraint (identified by the given identifier) that limits the enzyme
+requirements of the listed fluxes to the given limit (mass/gDW units).
+
+Note: [`simplified_enzyme_constraints`](@ref) and [`enzyme_constraints`](@ref)
+differ in how the capacity bound(s) are formulated. For the former, fluxes are
+used, but for the latter, gene products are used directly.
+"""
+function simplified_enzyme_constraints(
+    constraints::C.ConstraintTree;
+    fluxes = constraints.fluxes,
+    fluxes_reverse = constraints.fluxes_reverse,
+    fluxes_forward = constraints.fluxes_forward,
+    reaction_isozymes::Dict{String,Dict{String,Isozyme}},
+    gene_product_molar_masses::Dict{String,Float64},
+    capacity::Union{Vector{Tuple{String,Vector{String},Float64}},Float64},
+)
+    # get fastest isozyme for each reaction (flux * MW/kcat = isozyme mass required)
+    enzyme_speeds = Dict(
+        Symbol(rid) => (;
+            forward = minimum(
+                sum(
+                    stoich * gene_product_molar_masses[gid] for
+                    (gid, stoich) in iso.gene_product_stoichiometry
+                ) / iso.kcat_forward for iso in values(isos)
```

Comment:
do not divide

Reply:
hmm not sure its avoidable tbh. Somewhere you will have to divide.